python/ray/tune/pbt.py

```diff
@@ -2,189 +2,248 @@
 from __future__ import division
 from __future__ import print_function
 
-import collections
 import random
 import math
 import copy
 
+from ray.tune.error import TuneError
+from ray.tune.trial import Trial
 from ray.tune.trial_scheduler import FIFOScheduler, TrialScheduler
+from ray.tune.variant_generator import _format_vars
+
+
+# Parameters are transferred from the top PBT_QUANTILE fraction of trials to
+# the bottom PBT_QUANTILE fraction.
+PBT_QUANTILE = 0.25
+
+
+class PBTTrialState(object):
+    """Internal PBT state tracked per-trial."""
+
+    def __init__(self, trial):
+        self.orig_tag = trial.experiment_tag
+        self.last_score = None
+        self.last_checkpoint = None
+        self.last_perturbation_time = 0
+
+    def __repr__(self):
+        return str((
+            self.last_score, self.last_checkpoint,
+            self.last_perturbation_time))
+
+
+def explore(config, mutations, resample_probability):
+    """Return a config perturbed as specified.
+
+    Args:
+        config (dict) Original hyperparameter configuration.
```

Comment:
are colons needed here to show up on readthedocs? (`config (dict): `)

Reply:
Fixed